gears_cli/__main__.py

```diff
@@ -109,35 +132,35 @@ def run(host, port, password, requirements, filepath, extra_args):
             for i in range(len(errors)):
                 print(Colors.Bred('%d)\t%s' % (i + 1, str(errors[i]))))
 
-@gears_cli.command()
-@click.option('--host', default='localhost', help='Redis host to connect to')
-@click.option('--port', default=6379, type=int, help='Redis port to connect to')
-@click.option('--password', default=None, help='Redis password')
-@click.option('--save-directory', default='./', help='Directory for exported files')
-@click.argument('requirement')
-def export_requirement(host, port, password, save_directory, requirement):
-    r = create_connection(host, port, password, decode_responses=False);
-    
-    try:
-        metaDataValues, setializedData = r.execute_command('RG.PYEXPORTREQ', requirement)
-    except Exception as e:
-        print(Colors.Bred("failed exporting requirement (%s)" % str(e)))
-        exit(1)
-
-    metaData = {}
-    for i in range(0, len(metaDataValues), 2):
-        key = metaDataValues[i]
-        value = metaDataValues[i + 1]
+def extract_metadata(meta_data_reply):
+    meta_data = {}
+    for i in range(0, len(meta_data_reply), 2):
+        key = meta_data_reply[i]
+        value = meta_data_reply[i + 1]
```

Comment:
We can use something of the sort of:
```
def decode_utf(d):
    if isinstance(d, dict):
        return {k.decode('utf-8'): v.decode('utf-8') for k, v in d.items()}
    if isinstance(d, list):
        return [x.decode('utf-8') for x in d]
	if isinstance(d, bytes):
		return d.decode('utf-8')
    return d.decode('utf-8')
```

Reply:
I thought about this but then you need to call it recursively for each element inside the dict/list (if we want really a generic solution). So I thought for this specific case its enough.

Reply:
Just playing with it:
```
def decode_utf(d):
    if isinstance(d, dict):
        return {k.decode('utf-8'): decode_utf(v) for k, v in d.items()}
    if isinstance(d, list):
        return [decode_utf(x) for x in d]
    return d.decode('utf-8')
```
And we may optimize it by specifically asking about bytes before recursing.